poc/vdaf_mastic.py

```diff
@@ -71,10 +88,67 @@ def shard(cls, measurement, nonce, rand):
         )
 
         input_shares = [
-            (vidpf_init_seed[0], flp_leader_proof_share),
-            (vidpf_init_seed[1], flp_helper_proof_share_seed),
+            (vidpf_init_seed[0], flp_leader_proof_share, None),
+            (vidpf_init_seed[1], flp_helper_proof_share_seed, None),
         ]
         return (public_share, input_shares)
+    
+    @classmethod
+    def shard_with_joint_rand(cls, measurement, nonce, rand):
+        (vidpf_gen_rand, rand) = front(cls.Vidpf.RAND_SIZE, rand)
+        (flp_prove_rand_seed, rand) = front(cls.Xof.SEED_SIZE, rand)
+        (flp_helper_proof_share_seed, rand) = front(cls.Xof.SEED_SIZE, rand)
+        (flp_leader_blind, rand) = front(cls.Xof.SEED_SIZE, rand)
+        (flp_helper_blind, rand) = front(cls.Xof.SEED_SIZE, rand)
+
+        (alpha, meas) = measurement
+        beta = cls.Flp.encode(meas)
+
+        # Generate VIDPF keys.
+        (vidpf_init_seed, vidpf_correction_words, vidpf_cs_proofs) = \
+            cls.Vidpf.gen(alpha, beta, nonce, vidpf_gen_rand)
+        vidpf_public_share = (vidpf_correction_words, vidpf_cs_proofs)
+
+        # Generate FLP joint randomness.
+        joint_rand_parts = []
+        joint_rand_parts.append(cls.joint_rand_part(
+                0, flp_leader_blind, vidpf_init_seed[0], nonce))
+        joint_rand_parts.append(cls.joint_rand_part(
+                1, flp_helper_blind, vidpf_init_seed[1], nonce))
+        joint_rand = cls.joint_rand(
+            cls.joint_rand_seed(joint_rand_parts))
+        flp_public_share = joint_rand_parts
+        
+
+        # Generate FLP proof shares.
+        flp_prove_rand = cls.Xof.expand_into_vec(cls.Field,
+            flp_prove_rand_seed,
+            cls.domain_separation_tag(USAGE_PROVE_RAND),
+            b'',
+            cls.Flp.PROVE_RAND_LEN,
+        )
+
+        flp_proof = cls.Flp.prove(beta, flp_prove_rand, joint_rand)
+        flp_leader_proof_share = vec_sub(
+            flp_proof,
+            cls.helper_proof_share(flp_helper_proof_share_seed),
+        )
+
+        input_shares = [
+            (vidpf_init_seed[0], 
+             flp_leader_proof_share,
+             flp_leader_blind
+            ),
```

Comment:
I think something like this would be more Python-ish
```suggestion
            (
                vidpf_init_seed[0], 
                flp_leader_proof_share,
                flp_leader_blind,
            ),
```